io-engine/src/lvm/lv_replica.rs

```diff
@@ -0,0 +1,739 @@
+        })
+    }
+
+    /// List logical volumes using the provided options as query criteria.
+    /// All lv's are imported and all imports must succeed.
+    pub(crate) async fn list(
+        opts: &QueryArgs,
+    ) -> Result<Vec<LogicalVolume>, Error> {
+        let mut g_error = Ok(());
+        let mut lvs = Self::fetch(opts).await?;
+        for lv in &mut lvs {
+            match lv.import().await {
+                Ok(_) => {}
+                Err(error) => {
+                    tracing::error!(
+                        "Failed to import {vg_name}/{lv_name}: {error}",
+                        vg_name = lv.vg_name,
+                        lv_name = lv.lv_name,
+                    );
+                    g_error = Err(error);
+                }
+            }
+        }
+        g_error?;
+        Ok(lvs)
+    }
+
+    /// Fetch logical volumes using the provided options as query criteria.
+    async fn fetch(opts: &QueryArgs) -> Result<Vec<LogicalVolume>, Error> {
+        let mut args = vec![
+            "--report-format=json",
+            "--options=lv_name,lv_uuid,lv_size,lv_path,lv_tags,vg_name,vg_uuid,vg_tags,vg_extent_size",
+            "--units=b",
+            "--nosuffix",
+            "-q",
+        ];
+        let select = opts.query();
+        let select_query = format!("--select={select}");
+        if !select.is_empty() {
+            args.push(select_query.trim_end_matches(','));
+        }
+
+        let mut report: LogicalVolumeList =
+            LvmCmd::lv_list().args(args.as_slice()).report().await?;
+
+        report.lv.iter_mut().for_each(|lv| lv.import_attrs());
+
+        Ok(report.lv)
+    }
+
+    /// Destroy the logical volume.
+    /// This unloads the lvol from the Bdev module first and then proceeds to
+    /// remove the lv from the parent volume group.
+    pub(crate) async fn destroy(mut self) -> Result<(), Error> {
+        self.export_bdev().await?;
+        let ptpl = self.ptpl();
+        self.remove().await?;
+        ptpl.destroy().ok();
+        Ok(())
+    }
+
+    /// Import the Logical Volume.
+    /// This is a required step after creating the LV, which adds the name and
+    /// the share protocol as property tags.
+    /// The LV is then imported as an spdk BDEV, which allows it to be shared
+    /// via nvmf or open locally (ex: by the nexus).
+    pub(crate) async fn import(&mut self) -> Result<(), Error> {
+        self.import_bdev().await
+    }
+
+    /// Resize the logical volume AND the SPDK Bdev to the given size.
+    /// > Note: If the SPDK Bdev fails to be resized, then the LV size is
+    /// > reverted back.
+    /// # Warning: If the SPDK Bdev is not loaded, then it is also not resized
+    /// and the operation still succeeds.
+    pub(crate) async fn resize(&mut self, size: u64) -> Result<(), Error> {
+        let prev_size = self.size;
+        self.resize_lv(size).await?;
+
+        if let Err(error) = self.resize_bdev(size).await {
```

Comment:
How spdk resize is related to lvm resize here ? Both using different pool_type resource

Reply:
Not sure what you mean by both using different pool_type?
We put an spdk bdev on top of the lvm lv, so after resizing the lvm lv we have to resize the spdk bdev.